### PR TITLE
Add automation for cloudflare dns

### PR DESCRIPTION
**Pull request by MSevey**
# PULL REQUEST

## Overview
Add automation for cloudflare DNS.  Steps defined here:

https://github.com/SkynetLabs/skynet-webportal/blob/master/docker-compose.yml#L58

## Checklist

- [ ] All git commits are signed. (REQUIRED)
- [ ] All new methods or updated methods have clear docstrings.
- [ ] Testing added or updated for new methods.
- [ ] Verified if any changes impact the WebPortal Health Checks.
- [ ] Appropriate documentation updated.
- [ ] Changelog file created.

### CODE REVIEW

File: playbooks/tasks/portal-role-task-certbot-custom-dns.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+# Check for custom DNS and setup accordingly
+#
+# NOTE: currently the only custom setup that is handled is Cloudflare. Logic in
+# this file should be updated to only allow 1 custom config when others are
+# added.
+
+- name: Setup Cloudflare DNS for Certbot
+  include_tasks: tasks/portal-role-task-certbot-setup-cloudflare.yml
+  when: dns_cloudflare_api_token is defined
```
Comment (firyx) [inline playbooks/tasks/portal-role-task-certbot-custom-dns.yml:10]: Where is the secret value fo `dns_cloudflare_api_token` defined?
In hosts.ini or cluster yamls?
- this is not secured
=> it should be defined in secrets storage (cluster config) I think.

**Review by firyx — CHANGES_REQUESTED** [review]

**Review by firyx — CHANGES_REQUESTED** [review]

File: playbooks/tasks/portal-role-task.yml [hunk 2]
```diff
@@ -92,6 +92,12 @@
 - name: Include setting up MongoDB
   include_tasks: tasks/portal-role-task-mongo-setup.yml
 
+# Custom DNS Handling for Certbot
+# When the AWS keys are not set, check for other DNS configs for certbot.
+- name: Custom Certbot DNS
+  include_tasks: tasks/portal-role-task-certbot-custom-dns.yml
+  when: webportal_common_config.aws_access_key is not defined and webportal_common_config.aws_access_key is not defined
```
[the remark this answers is not in the document]
Comment (MSevey) [reply playbooks/tasks/portal-role-task.yml:99]: no, that is handled in the task. The task will check for various custom configs. So we just check that the default AWS is not defined here.

**Review by firyx — APPROVED** [review]

**Review by kwypchlo — COMMENTED** [review]
while this is improvements since we only currently supported route53, I think this should be more generic implementation, most of the certbot dns plugins require an ini file

check https://eff-certbot.readthedocs.io/en/stable/using.html#dns-plugins

it does not make sense to implement plugin specific settings for each one of them right?

**Comment by MSevey** [comment]
> while this is improvements since we only currently supported route53, I think this should be more generic implementation, most of the certbot dns plugins require an ini file
> 
> check https://eff-certbot.readthedocs.io/en/stable/using.html#dns-plugins
> 
> it does not make sense to implement plugin specific settings for each one of them right?

**Comment by MSevey** [comment]
> it does not make sense to implement plugin specific settings for each one of them right?

it depends.

There are going to be pieces that can be generic, and there will be pieces that I'm sure will need to be specific. We can't know that unless we go through the set up for each and every option. That is not something that I'm going to do for this PR, and it is why I set up the code the way I did.

This structure allows for handling future DNS plugins either as separate specific cases, or can be updated to be generic as needed.

Happy to make a follow up issue for future reference but I think the effort for making this generic right now is not worth the benefit.

**Review by kwypchlo — APPROVED** [review]

**Pull request merged**